node/src/store/indexed_db_store.rs

```diff
@@ -0,0 +1,514 @@
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use celestia_types::hash::Hash;
+use celestia_types::ExtendedHeader;
+use rexie::{Direction, Index, KeyRange, ObjectStore, Rexie, TransactionMode};
+use send_wrapper::SendWrapper;
+use serde::{Deserialize, Serialize};
+use serde_wasm_bindgen::{from_value, to_value};
+
+use crate::store::{Result, Store, StoreError};
+
+const DB_VERSION: u32 = 1;
+const HEADER_STORE_NAME: &str = "headers";
+const HASH_INDEX_NAME: &str = "hash";
+const HEIGHT_INDEX_NAME: &str = "height";
+
+#[derive(Debug, Serialize, Deserialize)]
+struct ExtendedHeaderEntry {
+    height: u64,
+    hash: Hash,
+    header: ExtendedHeader,
+}
+
+// SendWrapper usage is safe in wasm because we're running on single thread
+#[derive(Debug)]
+pub struct IndexedDbStore(Arc<SendWrapper<Rexie>>);
+
+impl IndexedDbStore {
+    pub async fn new_with_name(name: &str) -> rexie::Result<Self> {
+        let rexie = Rexie::builder(name)
+            .version(DB_VERSION)
+            .add_object_store(
+                ObjectStore::new(HEADER_STORE_NAME)
+                    .key_path("id")
+                    .auto_increment(true)
+                    .add_index(Index::new(HASH_INDEX_NAME, "hash").unique(true))
+                    .add_index(Index::new(HEIGHT_INDEX_NAME, "height").unique(true)),
+            )
+            .build()
+            .await?;
+        Ok(Self(Arc::new(SendWrapper::new(rexie))))
+    }
+
+    pub async fn new() -> rexie::Result<Self> {
+        Self::new_with_name("CELESTIA").await
+    }
+
+    pub async fn clear(self) -> rexie::Result<()> {
+        Rexie::delete(&self.0.name()).await
+    }
```

Comment:
Renamed to `delete_db` since what's that it does. And [indeed, we should](https://github.com/devashishdxt/rexie/issues/32). Fixed this method and added tests for it.